### PR TITLE
[INLONG-10381][Sort] Kafka source connector report audit attach input time

### LINKED ISSUE #10381: [Improve][Sort] Kafka source connector report audit attach input time
### Description

Kafka source connector not attach input time  when send audit. It will due to the report audit time not same with input time.

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XiaoYou201**
### [INLONG-10381][Sort] Kafka source connector report audit attach input time

Fixes #10381

### Motivation

Kafka source connector not attach input time when send audit. It will due to the report audit time not same with input time

### Modifications

Make DeserializationSchema report audit attach input time when deserialize.

### CODE REVIEW

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java [hunk 1]
```diff
@@ -44,6 +44,8 @@
  * */
 class DynamicKafkaDeserializationSchema implements KafkaDeserializationSchema<RowData> {
 
+    private static final int METADATA_CONSUME_TIME_INDEX = 0;
```
Comment (EMsnap) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java:47]: consume_time_index may not be 0

Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java]: fix it

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java [hunk 2]
```diff
@@ -131,12 +140,16 @@ public void deserialize(ConsumerRecord<byte[], byte[]> record, Collector<RowData
...
         }
-
         // project output while emitting values
         outputCollector.inputRecord = record;
         outputCollector.physicalKeyRows = keyCollector.buffer;
-        outputCollector.outputCollector =
-                sourceMetricData == null ? collector : new MetricsCollector<>(collector, sourceMetricData);
+        if (sourceMetricData != null) {
+            MetricsCollector<RowData> metricsCollector = new MetricsCollector<>(collector, sourceMetricData);
+            metricsCollector.resetTimestamp((Long) outputCollector.metadataConverters[consumeTimeIndex].read(record));
```
Comment (EMsnap) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java:148]: please consider the situation where you cannot get a consumeTimeIndex

Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java]: fix it

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java [hunk 3]
```diff
@@ -131,12 +140,21 @@ public void deserialize(ConsumerRecord<byte[], byte[]> record, Collector<RowData
...
         }
-
         // project output while emitting values
         outputCollector.inputRecord = record;
         outputCollector.physicalKeyRows = keyCollector.buffer;
-        outputCollector.outputCollector =
-                sourceMetricData == null ? collector : new MetricsCollector<>(collector, sourceMetricData);
+        if (sourceMetricData != null) {
+            MetricsCollector<RowData> metricsCollector = new MetricsCollector<>(collector, sourceMetricData);
+            if (consumeTimeIndex != -1) {
```
Comment (EMsnap) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java:148]: please extract method for consumeTimeIndex and getting time

Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java]: fix it, thx~

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java [hunk 4]
```diff
@@ -146,6 +160,14 @@ public void deserialize(ConsumerRecord<byte[], byte[]> record, Collector<RowData
         keyCollector.buffer.clear();
     }
 
+    private Long getRecordTime(int consumeTimeIndex, MetadataConverter[] metadataConverters,
```
[the remark this answers is not in the document]
Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java]: fix it, thx~

**Review by vernedeng — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**